packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -147,6 +140,23 @@ export class SelectedNetworkController extends BaseControllerV2<
     domain: Domain,
     networkClientId: NetworkClientId,
   ) {
+    const networkClient = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      networkClientId,
+    );
+    const networkProxy = this.#proxies.get(domain);
+    if (networkProxy === undefined) {
+      this.#proxies.set(domain, {
+        provider: createEventEmitterProxy(networkClient.provider),
```

Comment:
are proxy wrappers necessary if they're key'ed by domain already?

Reply:
Yes, since the provider for a  domain can change, we want to pass around 'sure to be updated accordingly' providers.